lib/AstToCStar.ml

```diff
@@ -219,6 +219,127 @@ let rec unit_to_void env e es extra =
   | _ ->
       CStar.Call (mk_expr env e, List.map (mk_expr env) es @ extra)
 
+(* This function focuses on the compilation of arithmetic expressions to take
+  into account implicit promotions to `int` (per the C standard) of types
+  smaller than int. We make the following assumption: int = 4 bytes, short = 2
+  bytes, char = 1 byte, and the unsigned versions of theses types are called
+  uint32_t, uint16_t, and uint8_t. Yes, the C standard doesn't impose these
+  choices, but all systems that we care about obey these restrictions
+  (https://www.viva64.com/en/t/0012/) and no, we don't intend to run the
+  generated C code on a PDP-11.
```

Comment:
It would be nice to document those hypotheses in, say, the Low* tutorial.
Thanks Jonathan!

Reply:
done in 2640744